### DISCUSSION TITLE
[Feature] API Keys for third party applications

### DISCUSSION THREAD

**Discussion by joeShuff** [immich-app/immich#4131] [Feature Request]
### The feature

I've looked into the possibility of creating a kind of digital photo frame with the source of the images being Immich. My plan was to create something for elderly family members who don't have easy access to technology.

Upon looking at the API documentation I think it'd be doable if we could create API keys with specific access.

The more granular control over an API key's access the better, but it could be iterated on.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
I mean, you can already create keys for you use case. The key is scoped to the user who created it, but you are right that there is no way to specify granular controls at the moment. That is more complicated and will require additional refactoring/standardization around our internal access controls.

**Comment by bo0tzz** [comment]
Closing in favour of #6910
